advtraj/integrate/forward.py

```diff
@@ -66,7 +66,10 @@ def _pt_ds_to_arr(ds_pt):
     def _pt_arr_to_ds(arr_pt):
         ds_pt = xr.Dataset()
         for n, c in enumerate("xyz"):
-            ds_pt[c] = arr_pt[n]
+            ds_pt[c] = xr.DataArray(arr_pt.reshape((3, -1))[n])
```

Comment:
Hmm, this looks more messy :) Also, doing a reshape three times to only extract one row each time.

Maybe something like

```python
ds_pt = xr.Dataset(dims=("trajectory_number"))
for n, c in enumerate("xyz"):
    ds_pt[c] = xr.DataArray(arr_pt[:,n], dims=("trajectory_number"))
```

would work?

Reply:
I'll look at this, but there was a problem when only a single trajectory - arr_pt then doesn't have the extra dimension. 
(I wasn't too concerned about repetition - reshape only changes a view of the data, not the data itself). However, I'll see if this can be made more 'xarray'.

Reply:
The problem is, indeed, with a single trajectory. I have changed this to:
```
        # Ensure arr_pt has 2 dimensions.
        arr_pt = arr_pt.reshape((3,-1))
        ds_pt = xr.Dataset()
        for n, c in enumerate("xyz"):
            ds_pt[c] = xr.DataArray(arr_pt[n,:], dims=("trajectory_number"))
```